src/sagas/productSaga.js

```diff
@@ -96,10 +123,32 @@ export function* setLabelsByCategory({ payload }) {
 }
 
 export function* getProductsByCategory({ payload }) {
-  const { category } = payload;
+  const { newCategory } = payload;
+  const product = yield select(getProduct);
+  const { category, typeList, brandList, rating, priceRange, page } = product;
+
   try {
-    const products = yield getProducts({ category });
-    yield put(getProductsSuccess({ products, category }));
+    if (newCategory) {
+      const products = yield getProducts({
+        category: newCategory,
+        typeList,
+        brandList,
+        rating,
+        priceRange,
+        page,
+      });
```

Comment:
```suggestion
      const products = yield getProducts({
        category: newCategory,
        ...rest
      });
```
Thử như này xem sao anh nhé